packages/aws-cdk-lib/aws-s3/lib/notifications-resource/notifications-resource.ts

```diff
@@ -133,6 +133,7 @@ export class BucketNotifications extends Construct {
           BucketName: this.bucket.bucketName,
           NotificationConfiguration: cdk.Lazy.any({ produce: () => this.renderNotificationConfiguration() }),
           Managed: managed,
+          S3NotificationsDeleteFeatureFlagEnabled: true
```

Comment:
This is not a feature flag if you hardcode it to `true`. Please follow this documentation to create the feature flag appropriately https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md#feature-flags. 

Alternatively if you can expose this property to users and make the default to be `false` and allow users to opt-in, that's also viable.